src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -658,81 +625,154 @@ private static ISymbol GetEventAccessor(IEventSymbol iEvent, IOperation operatio
+            static void CheckWithCallSiteAttributes(IOperation operation, OperationAnalysisContext context, PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>,
+                SmallDictionary<string, PlatformAttributes>> platformSpecificOperations, ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers,
+                ImmutableArray<string> msBuildPlatforms, ISymbol symbol, SmallDictionary<string, PlatformAttributes> operationAttributes)
             {
-                if (TryGetOrCreatePlatformAttributes(symbol, platformSpecificMembers, out var operationAttributes))
+                if (TryGetOrCreatePlatformAttributes(context.ContainingSymbol, platformSpecificMembers, out var callSiteAttributes))
                 {
-                    if (TryGetOrCreatePlatformAttributes(context.ContainingSymbol, platformSpecificMembers, out var callSiteAttributes))
+                    if (IsNotSuppressedByCallSite(operationAttributes, callSiteAttributes, msBuildPlatforms, out var notSuppressedAttributes))
                     {
-                        if (IsNotSuppressedByCallSite(operationAttributes, callSiteAttributes, msBuildPlatforms, out var notSuppressedAttributes))
-                        {
-                            platformSpecificOperations.TryAdd(operation, notSuppressedAttributes);
-                        }
+                        platformSpecificOperations.TryAdd(new KeyValuePair<IOperation, ISymbol>(operation, symbol), notSuppressedAttributes);
                     }
-                    else
+                }
+                else
+                {
+                    if (TryCopyAttributesNotSuppressedByMsBuild(operationAttributes, msBuildPlatforms, out var copiedAttributes))
                     {
-                        if (TryCopyAttributesNotSuppressedByMsBuild(operationAttributes, msBuildPlatforms, out var copiedAttributes))
-                        {
-                            platformSpecificOperations.TryAdd(operation, copiedAttributes);
-                        }
+                        platformSpecificOperations.TryAdd(new KeyValuePair<IOperation, ISymbol>(operation, symbol), copiedAttributes);
                     }
                 }
             }
-        }
 
-        private static bool TryCopyAttributesNotSuppressedByMsBuild(SmallDictionary<string, PlatformAttributes> operationAttributes,
-            ImmutableArray<string> msBuildPlatforms, out SmallDictionary<string, PlatformAttributes> copiedAttributes)
-        {
-            copiedAttributes = new SmallDictionary<string, PlatformAttributes>(StringComparer.OrdinalIgnoreCase);
-            foreach (var (platformName, attributes) in operationAttributes)
+            static bool TryGetOrCreateImmediatePlatformAttributes(ISymbol symbol, ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers,
```

Comment:
Why do we need a separate method `TryGetOrCreateImmediatePlatformAttributes`? Can we instead just add a new bool, say `immediateOnly` to `TryGetOrCreatePlatformAttributes` such that [this part](https://github.com/dotnet/roslyn-analyzers/blob/03a1c1666000aad58d660db44806710f7e56326c/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs#L968-L986) of that method is executed only when this bool is false?

Reply:
We can do that, for me it was more descriptive, but yes it is causing more noise in the PR, i will revert some of the changes